File: src/my_project/easy_problems/from101to150/matrix_cells_distance_order.py
from typing import List 
from collections import defaultdict
import math 

class Solution:
    def allCellsDistOrder(self, rows: int, cols: int, rCenter: int, cCenter: int) -> List[List[int]]:

        temp_collections = defaultdict(int)
        for i in range(rows):
            for j in range(cols):
                temp_collections[(i,j)] = abs(i-rCenter) + abs(j-cCenter)

        l = [([k[0],k[1]],v) for k,v in temp_collections.items()]
        l.sort(key = lambda x: x[1])
        l = [item[0] for item in l]
        return l


solution = Solution()

print(solution.allCellsDistOrder( rows = 2, cols = 3, rCenter = 1, cCenter = 2))

print(solution.allCellsDistOrder(rows = 1, cols = 2, rCenter = 0, cCenter = 0))

print(solution.allCellsDistOrder(rows = 2, cols = 2, rCenter = 0, cCenter = 1))

File: src/my_project_typescript/easy_problems_typescript/from1to50/matrixCellsDistanceOrder.ts
function allCellsDistOrder(rows: number, cols: number, rCenter: number, cCenter: number): number[][] {

    function distance(row:number, col:number): number{
        return Math.abs(row-rCenter) + Math.abs(col-cCenter)
    }

    let answer: number[][] = []
    for (let i = 0; i < rows; i++){
        for (let j = 0; j < cols; j++){
            answer.push([i,j])
        }
    }

    answer = answer.sort((a,b) => distance(a[0],a[1])<= distance(b[0],b[1])?-1:1 )
    
    return answer
};

// one function

function newFunctin(myVariable: number){
    return myVariable
}
allCellsDistOrder( 2,  2,  0,  1)
allCellsDistOrder( 2, 3, 1, 2)


